### PR TITLE
2018.05 Version 2.9.0-preview

### PR DESCRIPTION
**Pull request by XiaoningLiu**
2018.05 Version 2.9.0-preview

ALL
* Updated storage service version to 2017-11-09.
* Added `progress` event for `SpeedSummary` class, which will be triggered when every progress updates.
* Bumped version of request module from ~2.83.0 to ^2.86.0 to solve a vulnerability issue.

BLOB
* Added `createBlobServiceWithTokenCredential()` to create `BlobService` object with bearer tokens such as OAuth access token (in preview).
* Added support for '$web' as a valid blob container name for static website.
* Added support for write-once read-many containers.
* The `Get Container Properties` and `List Containers` APIs now return two new properties indicating whether the container has an immutability policy or a legal hold.
* The `Get Blob Properties` and `List Blobs` APIs now return the creation time of the blob as a property.

QUEUE
* Added `createQueueServiceWithTokenCredential()` to create `QueueService` object with bearer tokens such as OAuth access token (in preview).

### CODE REVIEW

File: browser/azure-storage.blob.export.js [hunk 1]
```diff
@@ -62,6 +66,7 @@ module.exports.Logger = azureCommon.Logger;
 module.exports.WebResource = azureCommon.WebResource;
 module.exports.Validate = azureCommon.validate;
 module.exports.date = azureCommon.date;
```
Comment (jiacfan) [inline browser/azure-storage.blob.export.js]: Why module.exports.date is in lowercase?

Comment (XiaoningLiu) [reply browser/azure-storage.blob.export.js]: Keep align (non-breaking) with Node.js exporting which is history code. https://github.com/Azure/azure-storage-node/blob/692ac1edb2dd01cb9bfd486b2cf89d7aa7e1e76c/lib/azure-storage.js#L364

File: browser/ChangeLog.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+ALL
+* Generated browser compatible JavaScript files based on Microsoft Azure Storage SDK for Node.js 2.9.0-preview.
+* Updated storage service version to 2017-11-09.
+* Added `/* eslint-disable */` to generated JS files to avoid eslint warnings when using with create-react-app.
+* Added `progress` event for `SpeedSummary` class, which will be triggered when every progress updates.
+
+BLOB
+* Add ```createBlobServiceWithBearerToken()``` to create ```BlobService``` object with bearer tokens such as OAuth access token.
+* Added support for '$web' as a valid blob container name for static website.
+* Added support for write-once read-many containers.
```
Comment (jiacfan) [inline browser/ChangeLog.md]: What about the following two changes in service version 2017-11-09?

The Get Container Properties and List Containers APIs now return two new properties indicating whether the container has an immutability policy or a legal hold. These policies are available in preview.
The Get Blob Properties and List Blobs APIs now return the creation time of the blob as a property.

File: browser/ChangeLog.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+2018.05 Version 2.9.100-preview
+
+ALL
+* Generated browser compatible JavaScript files based on Microsoft Azure Storage SDK for Node.js 2.9.0-preview.
+* Updated storage service version to 2017-11-09.
+* Added `/* eslint-disable */` to generated JS files to avoid eslint warnings when using with create-react-app.
+* Added `progress` event for `SpeedSummary` class, which will be triggered when every progress updates.
+
+BLOB
+* Add ```createBlobServiceWithBearerToken()``` to create ```BlobService``` object with bearer tokens such as OAuth access token.
```
Comment (jiacfan) [inline browser/ChangeLog.md]: Seems changed to createBlobServiceWithTokenCredential(), and same for Queue?

File: browser/ChangeLog.md [hunk 4]
```diff
@@ -1,5 +1,21 @@
 Note: This is the change log file for Azure Storage JavaScript Client Library.
 
+2018.05 Version 2.9.100-preview
+
+ALL
+* Generated browser compatible JavaScript files based on Microsoft Azure Storage SDK for Node.js 2.9.0-preview.
+* Updated storage service version to 2017-11-09.
+* Added `/* eslint-disable */` to generated JS files to avoid eslint warnings when using with create-react-app.
+* Added `progress` event for `SpeedSummary` class, which will be triggered when every progress updates.
```
Comment (jiacfan) [inline browser/ChangeLog.md]: What about the bug fix to GetBlob/GetFile

File: lib/common/services/storageservicesettings.js [hunk 5]
```diff
@@ -26,13 +26,15 @@ var StorageServiceClientConstants = Constants.StorageServiceClientConstants;
...
 var SR = require('../util/sr');
+var TokenCredential = require('../models/tokencredential');
 
 var useDevelopmentStorageSetting = ServiceSettings.setting(ConnectionStringKeys.USE_DEVELOPMENT_STORAGE_NAME, true);
 var developmentStorageProxyUriSetting = ServiceSettings.settingWithFunc(ConnectionStringKeys.DEVELOPMENT_STORAGE_PROXY_URI_NAME, Validate.isValidUri);
 var defaultEndpointsProtocolSetting = ServiceSettings.setting(ConnectionStringKeys.DEFAULT_ENDPOINTS_PROTOCOL_NAME, 'http', 'https');
 var accountNameSetting = ServiceSettings.setting(ConnectionStringKeys.ACCOUNT_NAME_NAME);
 var accountKeySetting = ServiceSettings.settingWithFunc(ConnectionStringKeys.ACCOUNT_KEY_NAME, Validate.isBase64Encoded);
 var sasSetting = ServiceSettings.settingWithFunc(ConnectionStringKeys.SHARED_ACCESS_SIGNATURE_NAME, _.isString);
+var tokenSetting = ServiceSettings.settingWithFunc('token', function (object) {return object instanceof TokenCredential;});
```
Comment (jiacfan) [inline lib/common/services/storageservicesettings.js]: [minor] Would better define token constant, not blocking.

File: lib/common/signing/sharedkey.js [hunk 6]
```diff
@@ -162,7 +162,7 @@ SharedKey.prototype._generateAccountSharedAccessSignature = function(sharedAcces
 /**
 * Signs a request with the Authentication header.
 *
-* @param {WebResource} The webresource to be signed.
+* @param {WebResource} webResource The webresource to be signed.
```
Comment (jiacfan) [inline lib/common/signing/sharedkey.js]: Aligning with other parameter.

Comment (jiacfan) [reply lib/common/signing/sharedkey.js]: Same in tokensigner.js

---
In reply to: [191098584](https://github.com/Azure/azure-storage-node/pull/453#discussion_r191098584) [](ancestors = 191098584)

File: lib/common/services/storageservicesettings.js (hunk 5 above)
Comment (XiaoningLiu) [reply lib/common/services/storageservicesettings.js]: 'token' is not part of `ConnectionStringKeys`. It's a design choice, let's we discuss it after this release for a better solution.

File: browser/ChangeLog.md (hunk 3 above)
Comment (XiaoningLiu) [reply browser/ChangeLog.md]: Good catch! Fixed in another commit

File: browser/ChangeLog.md (hunk 2 above)
Comment (XiaoningLiu) [reply browser/ChangeLog.md]: Updated.

File: lib/common/signing/sharedkey.js (hunk 6 above)
Comment (XiaoningLiu) [reply lib/common/signing/sharedkey.js]: aligned

File: browser/ChangeLog.md (hunk 4 above)
Comment (XiaoningLiu) [reply browser/ChangeLog.md]: As this comment was closed, no need to change.

**Pull request merged**